src/ut/realmmanager_test.cpp

```diff
@@ -465,3 +522,196 @@ TEST_F(RealmmanagerTest, SRVPriorityNegative)
 
   delete realm_manager;
 }
+
+TEST_F(RealmmanagerTest, PeerConnectionAlarm)
```

Comment:
This is a massive UT that's testing lots of different cases. Can you split these out into separate tests, and comment on each test what it covers. Commonising the AddrInfo creation would make sense as well.

You also need tests where:
* A peer fails multiple times - what happens to the time stored in the failed_peers map?
* Not all failed peers are aged out - e.g. have 5 failed peers and age out the 2nd and 4th.